### PR TITLE
feat: add rust codes for avl tree

### PR DESCRIPTION
**Pull request by night-cruise**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

The console outputs:
```
插入节点 1 后，AVL 树为
——— 1

插入节点 2 后，AVL 树为
    /——— 2
——— 1

插入节点 3 后，AVL 树为
    /——— 3
——— 2
    \——— 1

插入节点 4 后，AVL 树为
        /——— 4
    /——— 3
——— 2
    \——— 1

插入节点 5 后，AVL 树为
        /——— 5
    /——— 4
   |    \——— 3
——— 2
    \——— 1

插入节点 8 后，AVL 树为
        /——— 8
    /——— 5
——— 4
   |    /——— 3
    \——— 2
        \——— 1

插入节点 7 后，AVL 树为
        /——— 8
    /——— 7
   |    \——— 5
——— 4
   |    /——— 3
    \——— 2
        \——— 1

插入节点 9 后，AVL 树为
            /——— 9
        /——— 8
    /——— 7
   |    \——— 5
——— 4
   |    /——— 3
    \——— 2
        \——— 1

插入节点 10 后，AVL 树为
            /——— 10
        /——— 9
       |    \——— 8
    /——— 7
   |    \——— 5
——— 4
   |    /——— 3
    \——— 2
        \——— 1

插入节点 6 后，AVL 树为
            /——— 10
        /——— 9
       |    \——— 8
    /——— 7
   |   |    /——— 6
   |    \——— 5
——— 4
   |    /——— 3
    \——— 2
        \——— 1

插入节点 7 后，AVL 树为
            /——— 10
        /——— 9
       |    \——— 8
    /——— 7
   |   |    /——— 6
   |    \——— 5
——— 4
   |    /——— 3
    \——— 2
        \——— 1

删除节点 8 后，AVL 树为
            /——— 10
        /——— 9
    /——— 7
   |   |    /——— 6
   |    \——— 5
——— 4
   |    /——— 3
    \——— 2
        \——— 1

删除节点 5 后，AVL 树为
            /——— 10
        /——— 9
    /——— 7
   |    \——— 6
——— 4
   |    /——— 3
    \——— 2
        \——— 1

删除节点 4 后，AVL 树为
        /——— 10
    /——— 9
   |    \——— 7
——— 6
   |    /——— 3
    \——— 2
        \——— 1

查找到的节点对象为 TreeNode { val: 7, high: 0, parent: None, left: None, right: None }，节点值 = 7
```

### CODE REVIEW

File: codes/rust/chapter_tree/avl_tree.rs [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+                    Ordering::Less => {
+                        let right = node.borrow().right.clone();
+                        node.borrow_mut().right = Self::insert_helper(right, val);
+                    }
+                    Ordering::Equal => {
+                        return Some(node); // 重复节点不插入，直接返回
+                    }
+                }
+                Self::update_height(Some(node.clone())); // 更新节点高度
+                                                         /* 2. 执行旋转操作，使该子树重新恢复平衡 */
```
Comment (krahets) [inline codes/rust/chapter_tree/avl_tree.rs:160]: Please fix indentation

File: codes/rust/chapter_tree/avl_tree.rs [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+                            break;
+                        }
+                        temp = temp_left.unwrap();
+                    }
+                    let right = node.borrow().right.clone();
+                    node.borrow_mut().right = Self::remove_helper(right, temp.borrow().val);
+                    node.borrow_mut().val = temp.borrow().val;
+                }
+                Self::update_height(Some(node.clone())); // 更新节点高度
+                                                         /* 2. 执行旋转操作，使该子树重新恢复平衡 */
```
Comment (krahets) [inline codes/rust/chapter_tree/avl_tree.rs:214]: Please fix indentation

File: codes/rust/include/tree_node.rs [hunk 3]
```diff
@@ -9,6 +9,7 @@ use std::collections::VecDeque;
 use std::rc::Rc;
 
 #[allow(dead_code)]
+#[derive(Debug)]
 pub struct TreeNode {
     pub val: i32,
     pub high: i32,
```
Comment (krahets) [inline codes/rust/include/tree_node.rs:15]: height

**Review by krahets — COMMENTED** [review]
Please address the comment, thx

File: codes/rust/include/tree_node.rs (hunk 3 above)
Comment (night-cruise) [reply codes/rust/include/tree_node.rs:15]: The `high` field is not defined by me, is it ok to be modified?

Comment (krahets) [reply codes/rust/include/tree_node.rs:15]: Yes, every file can be modified, especially found something incorrect. All the changes will be reviewed in PR.

Comment (krahets) [reply codes/rust/include/tree_node.rs:15]: Actually, welcome to pay more attention to the public module or others' code.

**Review by krahets — APPROVED** [review]
Thanks for the clean code!

**Pull request merged**